pkg/services/ngalert/store/alertmanager.go

```diff
@@ -19,6 +19,10 @@ var (
 	// ErrVersionLockedObjectNotFound is returned when an object is not
 	// found using the current hash.
 	ErrVersionLockedObjectNotFound = fmt.Errorf("could not find object using provided id and hash")
+	// ConfigRecordsLimit defines the limit of how many alertmanager configuration versions
+	// should be stored in the database for each organization including the current one.
+	// Has to be > 0
+	ConfigRecordsLimit int64 = 100
```

Comment:
After the previous incident, I wonder if we should first delete all _sequential_ configs with the same hash (except one for each unique hash)?

That way, we are always properly preserving history rather than just 100 duplicates. Otherwise, the duplication bug makes it so everyone's history is wiped. If this is too difficult to implement it's probably not worth worrying about for now - reducing DB size is more urgent.

Reply:
I thought about it, but it will automatically heal over time. Implementing it is way too much effort considering the return.